helpers/templating.js

```diff
@@ -3,5 +3,6 @@ export default function parseTemplateString(string, variables) {
     return string
   }
   const searchTerm = /<<([^>]*)>>/g // "<<myVariable>>"
-  return decodeURI(encodeURI(string)).replace(searchTerm, (match, p1) => variables[p1] || "")
+  string = new URL(string).toString();
```

Comment:
This seems to be a breaking change.
Can you assign a new flag for storing `string` instead of mutating it with the original parameter.

Deploy preview: https://deploy-preview-1453--hoppscotch.netlify.app (check console logs)
Refer: https://stackoverflow.com/questions/518000/is-javascript-a-pass-by-reference-or-pass-by-value-language

Reply:
@liyasthomas explored the error some more and it turned out the error was because of the interdependency between url and uri (and hence a recursive call of getters). Generally an update in uri leads to an update in url but on startup we give url and calculate uri from url. Now I am making sure url is depends on the value of uri and not vice versa. Please have a look and let me know if it seems fine now.

Reply:
Ahh, well, here's the [deploy preview](https://deploy-preview-1453--hoppscotch.netlify.app). It's failing to construct URL object from string.

Reply:
@liyasthomas I think you'd have to open the preview in incognito, the link you shared has an empty request appended to it. The [preview](https://deploy-preview-1453--hoppscotch.netlify.app/) is working fine from my side.

Reply:
Can you share a screenshot of the response section with the default request set up?

`GET https://httpbin.org/get`

Reply:
@liyasthomas I see what you meant, sorry. I assumed parseTemplateString is used only to parse URLs. It's breaking where the path is being passed.